test/jdk/javax/swing/JToolTip/4846413/bug4846413.java

```diff
@@ -28,71 +28,94 @@
  * @summary Checks if No tooltip modification when no KeyStroke modifier
  * @library ../../regtesthelpers
  * @build Util
- * @author Konstantin Eremin
  * @run main bug4846413
  */
+
+import java.io.File;
 import java.awt.Dimension;
 import java.awt.Point;
+import java.awt.Rectangle;
 import java.awt.Robot;
 import java.awt.Toolkit;
-import javax.swing.*;
-import java.awt.event.*;
+import java.awt.event.ContainerAdapter;
+import java.awt.event.ContainerEvent;
+import java.awt.event.KeyEvent;
+import java.awt.image.BufferedImage;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLayeredPane;
+import javax.swing.JToolTip;
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.imageio.ImageIO;
 import javax.swing.plaf.metal.MetalToolTipUI;
 
 public class bug4846413 {
 
     private static volatile boolean isTooltipAdded;
     private static JButton button;
+    private static JFrame frame;
+    private static Robot robot;
 
     public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoDelay(100);
 
-        Robot robot = new Robot();
-        robot.setAutoDelay(50);
+            UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
 
-        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+            SwingUtilities.invokeAndWait(() -> createAndShowGUI());
 
-        javax.swing.SwingUtilities.invokeAndWait(new Runnable() {
+            robot.waitForIdle();
+            robot.delay(1000);
 
-            public void run() {
-                createAndShowGUI();
+            Point movePoint = getButtonPoint();
+            robot.mouseMove(movePoint.x, movePoint.y);
+            if (System.getProperty("os.name").contains("OS X")) {
+                String version = System.getProperty("os.version", "");
+                if (version.startsWith("14.")) {
+                    robot.mouseMove(movePoint.x, movePoint.y);
```

Comment:
Adding to Phil's point, version specific test fixes would require to be removed and/or reworked if the issue gets fixed on a newer macOS version.